source/use-cases/sensitive-data-encryption.txt

```diff
@@ -13,26 +13,51 @@ Sensitive Data Encryption
-consectetur molestie pellentesque ac, blandit non libero.
+If you store information in MongoDB such as confidential personal details, payment information, or proprietary data, you are probably interested in protecting your data from unauthorized access. In addition, the data you store may be subject to governance, privacy, and security compliance mandates. 
+
+MongoDB offers several methods that reduce your risk of a data breach resulting from unauthorized access whether accidental or by a malicious act including:
+
+* `Role-based access control <https://docs.mongodb.com/manual/core/authorization/>`_
+* `Monitoring and logging <https://docs.mongodb.com/manual/core/auditing/>`_
+* `TLS/SSL network transport encryption <https://docs.mongodb.com/manual/core/security-transport-encryption/>`_
+* `Data encryption at rest <https://docs.mongodb.com/manual/core/security-encryption-at-rest/>`_
+* `Client-side field level encryption <https://docs.mongodb.com/manual/core/security-client-side-encryption/>`_
+
+This document guides you through setup and implementation of Client-Side Field Level Encryption (CSFLE), a feature that encrypts document fields specified by a developer, in a example scenario to demonstrate how it can be used to secure your data.
+
+.. note::
+
+  Client-side Field Level Encryption is available starting in MongoDB 4.2 Enterprise only
 
 Problem
 ~~~~~~~
 
-Pellentesque non mi et est congue mattis pretium sit amet mi. Praesent
-nec tortor nec arcu molestie dapibus. Phasellus at nunc ut quam molestie
-finibus. Suspendisse finibus lectus eget nulla eleifend, vel consequat
-leo rhoncus. Phasellus lacinia tortor id aliquet condimentum. Morbi
-sodales est a leo rhoncus, non pretium neque dignissim.
+In this scenario, we secure sensitive data on a Medical Care Management System which stores patients' personal information, appointments, and vital sign measurement reports for a fictitious company, *MedcoMD*. While none of this data should ever be exposed to the public, if a malicious user gained access to data such as the patient's social security number (SSN) or their vital sign measurements, it would be disastrous and damaging to the company.
+
+MedcoMD provides access to this system for two types of users:
+
+1.  Receptionists, who use the system to schedule appointments for Patients, identified by a combination of contact information and their SSN.
+2.  Doctors, who use the system to access and enter Patients' vital sign measurements
+
+While MedcoMD trusts their employees never to retrieve sensitive data, there are several ways it could be exposed if left unencrypted including by leaving it exposed on a screen or print-out, by sniffing the data from network packets on an insecure network, or by obtaining unauthorized access to the database. If they store complete, encrypted documents on the server, every query would require decrypting the entire collection which is too computationally expensive, and therefore, slow.
```

Comment:
nit: several ways _that_ it could...

Reply:
nit: we say "if left unencrypted" but we haven't quite introduced the case for encryption to the scenario yet. I wonder if it makes sense to say "these are some ways that data can be stolen" and then say "because of this they must have some data encryption but also it's too expensive to encrypt everything."

i.e. More or less what you already have here but phrased so that generic encryption follows as a solution to the problem.

Reply:
also is it "employees never retrieve sensitive data" or "employees only view sensitive data when it's appropriate"

Reply:
We need to be careful not to confuse full document encryption with encryption at rest. Maybe something like

> While encryption at rest protects against a malicious actor from gaining data on disk, if they were to compromise valid credentials they would have access like any other user. Full document encryption would prove too slow as there is a lot of information that doesn't need to be encrypted, and every query would ...

Reply:
@terakilobyte I'm concerned that talking about encryption at rest will require explanation and shift focus away from the Problem section. Could we chat in-person about this so that I can understand what you're looking for?

Reply:
Discussed in-person:
Will leave out explanation and describe use cases and risks in list format